### PR TITLE
RM-7517: Update RhinoMocks to CoreRhinoMocks

### PR DESCRIPTION
**Pull request by iam-mholle**
https://re-motion.atlassian.net/browse/RM-7517

### CODE REVIEW

File: Remotion/Core/Reflection.UnitTests/ReflectionExtensionsTest.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public void ConvertToRuntimeType_WithOtherITypeInformation_ThrowsInvalidOperatio
           Throws.InvalidOperationException.And.Message.EqualTo (
               string.Format (
                   "The type 'TheName' cannot be converted to a runtime type because no conversion is registered for '{0}'.",
-                  typeInformation.GetType().Name)));
+                  typeInformation.GetType().FullName)));
```
Comment (dukorama) [inline Remotion/Core/Reflection.UnitTests/ReflectionExtensionsTest.cs:119]: What's the motivation for this change? Is Name not specific enough?

File: Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs [hunk 2]
```diff
@@ -150,7 +150,7 @@ public void ExpectObjectSecurityStrategyHasAccess (SecurableObject securableObje
               Arg.Is (_stubUser),
               Arg<IReadOnlyList<AccessType>>.List.Equal (new[] { AccessType.Get (accessTypeEnum) })))
           .WhenCalled (mi => CheckTransaction())
-          .Do (doDelegate);
+          .Do (doDelegate).Repeat.Once();
     }
```
Comment (dukorama) [inline Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs:154]: Why do we need all these "Repeat.Once()" calls?

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs [hunk 3]
```diff
@@ -77,6 +77,7 @@ public override void SetUp ()
       MockRepository mockRepository = new MockRepository();
       _page = mockRepository.PartialMultiMock<Page> (typeof (ISmartPage));
       ((ISmartPage) _page).Stub (stub => stub.Context).Return (new HttpContextWrapper (HttpContext.Current));
+      ((ISmartPage) _page).Stub (stub => stub.Site).Return (null);
```
Comment (dukorama) [inline Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs:80]: Why is this necessary? Shouldn't be null the default for a partial mock even without stubbing?

**Review by dukorama — CHANGES_REQUESTED** [review]

File: Remotion/Core/Reflection.UnitTests/ReflectionExtensionsTest.cs (hunk 1 above)
Comment (iam-mholle) [reply Remotion/Core/Reflection.UnitTests/ReflectionExtensionsTest.cs:119]: The actual exception message changed, probably with the Castle upgrade in CoreRhinoMocks

File: Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs (hunk 2 above)
Comment (iam-mholle) [reply Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs:154]: This change is needed when upgrading from Rhino.Mocks 3.6.0 to 3.6.1.

Comment (dukorama) [reply Remotion/Data/DomainObjects.Security.UnitTests/SecurityClientTransactionExtensionTests/TestHelper.cs:154]: As discussed.

File: Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs (hunk 3 above)
Comment (iam-mholle) [reply Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs:80]: Rhino.Mocks behavior has changed between 3.6.0 & 3.6.1. If this stub is not set up, the following exception is thrown:
```
Rhino.Mocks.Exceptions.ExpectationViolationException : IComponent.get_Site(); Expected #0, Actual #1.
   at Rhino.Mocks.MethodRecorders.UnorderedMethodRecorder.DoGetRecordedExpectation(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.MethodRecorders.MethodRecorderBase.GetRecordedExpectation(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.ReplayPartialMockState.DoMethodCall(IInvocation invocation, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.ReplayMockState.MethodCall(IInvocation invocation, MethodInfo method, Object[] args)
   at Rhino.Mocks.MockRepository.MethodCall(IInvocation invocation, Object proxy, MethodInfo method, Object[] args)
   at Rhino.Mocks.Impl.Invocation.Actions.RegularInvocation.PerformAgainst(IInvocation invocation)
   at Rhino.Mocks.Impl.RhinoInterceptor.Intercept(IInvocation invocation)
   at Castle.DynamicProxy.AbstractInvocation.Proceed()
   at Castle.Proxies.PageProxy.get_Site()
   at Remotion.Web.Utilities.ControlHelper.IsDesignMode(IControl control)
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBinding.get_Property()
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBoundWebControl.get_Property()
   at Remotion.ObjectBinding.Web.UI.Controls.BusinessObjectBoundEditableWebControl.get_IsReadOnly()
   at Remotion.ObjectBinding.Web.UI.Controls.BocAutoCompleteReferenceValue.GetTrackedClientIDs()
   at Remotion.ObjectBinding.Web.UnitTests.UI.Controls.BocAutoCompleteReferenceValueTest.GetTrackedClientIDsInEditMode()
```

Comment (dukorama) [reply Remotion/ObjectBinding/Web.UnitTests/UI/Controls/BocAutoCompleteReferenceValueTest.cs:80]: Interesting...

**Review by dukorama — APPROVED** [review]

**Review by MichaelKetting — APPROVED** [review]

**Comment by iam-mholle** [comment]
Added to master with 15ac877355df7a26442d83db3eff9a6a53c7cf81.
